include/field.hxx

```diff
@@ -37,7 +37,11 @@ class Field;
 
 class Mesh;
 class Coordinates;
-extern Mesh * mesh; ///< Global mesh
+namespace bout {
+namespace globals {
+  extern Mesh* mesh; ///< Global mesh
```

Comment:
Is there a reason not to include `globals.hxx` here?

Reply:
I don't think so... At the moment including `globals.hxx` includes `bout/mesh.hxx`, `datafile.hxx` and `bout/macro_for_each.hxx`, but I guess we can just forward-declare `Mesh` and `Datafile`.

Reply:
ah, don't think we can forward-declare `Datafile` because `dump` is a variable, not a pointer. So including `globals.hxx` will have to include `datafile.hxx` too, which might not be necessary...

Reply:
Ah ok -- https://github.com/boutproject/BOUT-dev/pull/1464/commits/cd6929ce5f4e402bdfb90ce723bae48f28def63c provides another way to avoid this (as mentioned earlier). 

Similar(ish) changes to `bout_types.hxx`  in https://github.com/boutproject/BOUT-dev/pull/1464/commits/c5d37d4b6cc6c9f57fa9e2673f59fcb9a4946f48 (and bug fix in https://github.com/boutproject/BOUT-dev/pull/1464/commits/52e72e2405cf1163bb1a2542ffa650daaeab6e59) seem to have given a decent speedup in travis builds (maybe 1-2 minutes) which is motivating me to look into shifting more stuff to source rather than header where possible.

I guess this might be something for another time/PR